src/main/java/io/openliberty/tools/intellij/lsp4mp/lsp4ij/LanguageServerWrapper.java

```diff
@@ -636,6 +629,12 @@ private CompletableFuture<LanguageServer> connect(@Nonnull URI absolutePath, Doc
                 }
                 DocumentContentSynchronizer listener = new DocumentContentSynchronizer(this, theDocument, syncKind);
                 theDocument.addDocumentListener(listener);
+                Set<DocumentContentSynchronizer> synchronizers = theDocument.getUserData(DocumentContentSynchronizer.KEY);
+                if (synchronizers == null) {
+                    synchronizers = Collections.synchronizedSet(new LinkedHashSet<>());
+                    theDocument.putUserData(DocumentContentSynchronizer.KEY, synchronizers);
```

Comment:
This is just a work around for there being no get() method for listeners. Just needed a unique ID/key for the set of listeners that are attached to the Document so that they can be retrieved from the Document to force a refresh of the diagnostics when the config source cache is cleared.

Reply:
Maybe `DocumentContentSynchronizer.KEY` could have a different name to suggest these are alternate synchronizers e.g. `AltDocumentContentSynchronizer.KEY`. Somehow differentiating from the `listener`.

Reply:
My first idea wasn't so great. I should have suggested `SetKey` as a name or `ListenerSet` or `AltListeners`. Having a good name can show new people what the object is for.